yurt/yurt_core/add.py

```diff
@@ -209,5 +211,48 @@ def vault(dest):
     with open(dest_path, 'w') as outfile:
         json.dump(settings, outfile)
 
+
+@add.command()
+@click.option("--name", default=None, help='Ansible Role name')
+@click.option("--remote/--local", default=False, help='Install role from AnsibleGalaxy(remote) or generate one (local)')
+def role(**kwargs):
+    ROLE_ATTRIBUTE_MAPPING = OrderedDict([
+        ('name', 'What will you name this role?: '),
+        ('remote', 'Install existing role from Ansible Galaxy (Y/N)?: '),
+    ])
+    try:
+        question_items = ROLE_ATTRIBUTE_MAPPING.iteritems()
+    except AttributeError:
+        question_items = ROLE_ATTRIBUTE_MAPPING.items()
+
+    # TODO: Abstract this question-loop with a re-usable util function
+    settings = {}
+    for attribute, prompt in question_items:
+        if kwargs[attribute] is None:
+            settings[attribute] = raw_input_wrapper(prompt, True)
+        else:
+            settings[attribute] = kwargs[attribute]
+
+    project_path = find_project_folder()
+    path_to_roles = ['orchestration', 'roles']
+
+    if settings['remote']:
+        roles_path = os.path.join(project_path, *path_to_roles)
+        ansible_galaxy_cmd = 'ansible-galaxy install -p {} {}'.format(roles_path, settings["name"])
```

Comment:
throughout `yurt` so far you were sticking with using indexes in string formatting - do we want to keep that standard?